QRScanner/QRScannerView.swift

```diff
@@ -71,11 +74,25 @@ public class QRScannerView: UIView {
         focusImageView.removeFromSuperview()
         qrCodeImageView.removeFromSuperview()
         setupImageViews()
+        qrCodeImage = nil
         videoDataOutputEnable = false
         metadataOutputEnable = true
     }
 
+    public func setTorchActive(isOn: Bool) {
+        guard let videoDevice = AVCaptureDevice.default(for: .video),
+            videoDevice.hasTorch, videoDevice.isTorchAvailable else {
+                return
+        }
+        DispatchQueue.main.async {
+            try? videoDevice.lockForConfiguration()
+            videoDevice.torchMode = isOn ? .on : .off
+            videoDevice.unlockForConfiguration()
+        }
```

Comment:
[nits] I think use only DispatchQueue.main.async in AVCaptureMetadataOutputObjectsDelegate is simpler.
When source code from async thread to main thread,  we add DispatchQueue.main.async.

```
extension QRScannerView: AVCaptureMetadataOutputObjectsDelegate {

	DispatchQueue.main.async { [weak self] in
   		guard let strongSelf = self else { return }
                strongSelf.setTorchActive(isOn: false)
		strongSelf.moveImageViews(qrCode: stringValue, corners: readableObject.corners)
        }
}
```

Remove DispatchQueue.main.async

```
public func setTorchActive(isOn: Bool) {
        guard let videoDevice = AVCaptureDevice.default(for: .video),
            videoDevice.hasTorch, videoDevice.isTorchAvailable else {
                return
        }
        
        try? videoDevice.lockForConfiguration()
        videoDevice.torchMode = isOn ? .on : .off
        videoDevice.unlockForConfiguration()
    }

```

```
private func didChangeTorchActive(isOn: Bool) {
        isTorchActive = isOn
        delegate?.didChangeTorchActive(isOn: isOn)
    }
```

Reply:
>When source code from async thread to main thread, we add DispatchQueue.main.async.

Yes, this is exactly right.
Calling from `AVCaptureMetadataOutputObjectsDelegate` is from main thread.
But `setTorchActive(isOn:)` is called from outside of framework so I think dispatch main queue is required.

_But I also think calling thread is almost main thread because caller is parent view or parent view controller._
_What do you think about this? Remove `Dispatch.main.async` is better? I'm wondering 🤔_

Reply:
I means move DispatchQueue.main.async from setTorchActive func to AVCaptureMetadataOutputObjectsDelegate.

For now
```
extension QRScannerView: AVCaptureMetadataOutputObjectsDelegate {
	// do some thing in async thread (metadataQueue)
	setTorchActive(..)

	DispatchQueue.main.async {
		// call func in main thread
		moveImageViews(..)
	}
}

func setTorchActive() {
	DispatchQueue.main.async {
		// do some thing.
	}
}
```

I think like this is simpler.
```
extension QRScannerView: AVCaptureMetadataOutputObjectsDelegate {
	// do some thing in async thread (metadataQueue)

	DispatchQueue.main.async {
		// call func in main thread
		setTorchActive(..)
		moveImageViews(..)
	}

}

func setTorchActive() {
		// do some thing.
}
```

What do you think?

Reply:
I got it!
I think the change is better and I will fix it 👍